### DISCUSSION TITLE
lineage can not effective

### DISCUSSION THREAD

**Discussion by davidyuan1223** [apache/kyuubi#5429] [Q&A] [answered]
this is spark-submit params
![image](https://github.com/apache/kyuubi/assets/51512358/d9b1304d-19e0-4728-a941-41c24fa2398b)
the lineage is use, and jar is in the spark's jars directory,when execute sql `select a as col0, b as col1 from test_table0`,there is no lineage log print
![image](https://github.com/apache/kyuubi/assets/51512358/6325eeb7-59e4-4022-825d-743b924943f3)
how to resolve this problem?

**Comment by wForget** [comment, marked as answer]
The spark lineage plugin sends lineage events to spark event bus by default, without logging them. You need to implement EventListener to handle it or use kyuubi event system. See more: https://kyuubi.readthedocs.io/en/master/extensions/engines/spark/lineage.html#get-lineage-events

**Reply by davidyuan1223** [reply]
get, thanks!!!!
